libkbfs/md_journal.go

```diff
@@ -464,7 +464,10 @@ func (j *mdJournal) convertToBranch(
 		brmd.SetUnmerged()
 		brmd.SetBranchID(bid)
 
-		// Delete the old "merged" version from the cache.
+		// Delete the old "merged" version from the cache.  We aren't
+		// equipped here to cache the new version, but it will
+		// eventually get passed via an mdFlushListener to somewhere
+		// that will cache it.
```

Comment:
Actually, I think it might be easier to do it here. Rough outline:

1. Save a list of the md cache keys and new brmd while looping.
2. After the loop, for each entry:
2a. Look up the old cache entry. If it doesn't exist, continue.
2b. Otherwise, make a deep copy of it, and replace its brmd with the newly-generated one.
2c. Delete the old cache entry, put the new cache entry.

What do you think? The problem with the current approach is that the gap between dropping the old version and putting the new version is large enough that when we put the new version, we may be kicking out hotter cache entries. In this case, we'd be doing so to a lesser extent. Ideally, we'd be able to replace the entry in the MDCache LRU without changing the LRU info.